README.md

```diff
@@ -12,36 +12,23 @@ wither
 
 An ODM for MongoDB built upon the [mongo rust driver](https://github.com/mongodb-labs/mongo-rust-driver-prototype). Please ⭐ on [github](https://github.com/thedodd/wither)!
 
-The primary goal of this project is to provide a simple, sane & predictable interface into MongoDB based on data models. If at any point this system might get in your way, you have direct access to the underlying driver. This project is tested against MongoDB `3.2`, `3.4`, `3.6` & `4.0`.
+The primary goal of this project is to provide a simple, sane & predictable interface into MongoDB based on data models. If at any point this system might get in your way, you have direct access to the underlying driver. This project is tested against MongoDB `3.2`, `3.4`, `3.6`, `4.0` & `4.2`.
```

Comment:
The new driver only supports back to 3.6; are you sure that this still works with 3.2 and 3.4 after porting to the new driver?

Reply:
Nope. Not at all haha. Still deep in the guts of the refactor right now, doing lots of work on the derive system as time permits. 

On the old driver, I had a test matrix which was executing all of the tests in this system against the versions mentioned here. I will definitely update this info as this PR stabilizes. 

Thanks for the heads-up @saghm!

Reply:
@saghm I noticed the [latest release of the driver](https://github.com/mongodb/mongo-rust-driver/releases/tag/v0.11.0) has annotated [ErrorKind](https://github.com/mongodb/mongo-rust-driver/blob/3323ed8501b68b58951260f0edc47dfa20d36479/src/error.rs#L96) and its variants such as **ArgumentError** `#[non_exhaustive]`.

IMO I think this prevents extended projects for example **wither** from the flexibility of providing direct wrapping atop of the underlying driver i.e. validations etc. as required by the driver. 

What do you think?

Reply:
@losintikfos you bring up a good point. It would be nice to have an `ErrorKind::Other(String)` or the like. Something which we could use as a fallback in a library like this.

Otherwise, the only other real option is to just create a new error type in this lib which wraps a mongo error as one of its enum variants or the like.

Reply:
The reason we marked things as non-exhaustive in the driver is that we want to be able to commit to API stability. Without non-exhaustive, adding new fields or variants to types would be a breaking change. 

I actually wasn't aware that `wither` was constructing `mongodb` errors to return. Would it be too much of a burden to define your own error type?

Reply:
@saghm no, that's totally fine. It is already on my shortlist to knock out.